CHANGELOG.md

```diff
@@ -53,6 +54,17 @@
 
 * Improved the accuracy of location tracking and off-route detection. ([#2319](https://github.com/mapbox/mapbox-navigation-ios/pull/2319))
 * Added the `PassiveLocationManager` class for use with the `MGLMapView.locationManager` property. Unlike `CLLocationManager`, this class causes the map view to display user locations snapped to the road network, just like during turn-by-turn navigation. To receive these locations without an `MGLMapView`, use the `PassiveLocationDataSource` class and implement the `PassiveLocationDataSourceDelegate.passiveLocationDataSource(_:didUpdateLocation:rawLocation:)` method or observe `Notification.Name.passiveLocationDataSourceDidUpdate` notifications. ([#2410](https://github.com/mapbox/mapbox-navigation-ios/pull/2410))
+* The `NavigationViewController.route` and `NavigationService.route` properties are now read-only. To change the route that the user is traveling along, set the `NavigationViewController.indexedRoute` or `NavigationService.indexedRoute` property instead, pairing the route with the index of the route in the original `RouteResponse` object. ([#2366](https://github.com/mapbox/mapbox-navigation-ios/pull/2366))
+* The following methods now require a route index to be passed in as an argument ([#2366](https://github.com/mapbox/mapbox-navigation-ios/pull/2366)):
+   * `NavigationViewController(for:routeIndex:routeOptions:navigationOptions:)`
+   * `NavigationViewController(route:routeIndex:routeOptions:navigationService:)`
+   * `CarPlayManagerDelegate.carPlayManager(_:navigationServiceAlong:routeIndex:routeOptions:desiredSimulationMode:)`
+   * `MapboxNavigationService(route:routeIndex:routeOptions:)`
+   * `MapboxNavigationService(route:routeIndex:routeOptions:directions:locationSource:eventsManagerType:simulating:routerType:)`
+   * `RouteProgress(route:routeIndex:options:legIndex:spokenInstructionIndex:)`
+   * `RouteProgress(route:routeIndex:options:legIndex:spokenInstructionIndex:)`
+   * `Router(along:routeIndex:options:directions:dataSource:)`
+   * `RouteController(along:routeIndex:options:directions:dataSource:)`
```

Comment:
This blurb nicely encapsulates why I’m nervous about this PR. Adding these extra arguments is pretty safe from a runtime stability standpoint, other than in some of the CarPlay code that relies on casting `Any` into tuples, but it’s always a big deal to add complexity to the basic setup workflow in NavigationViewController particularly.

Reply:
If I'm not mistaken in all public API methods which were modified to contain `routeIndex` `RouteOptions` argument is also present. Would it work if we add `routeIndex` as part of options?

Reply:
A single RouteOptions may be associated with more than one route, so I don’t think that would work. Unless you mean NavigationOptions?